lecture/src/main/java/de/tum/in/www1/artemis/lecture/service/messaging/ArtemisConsumer.java

```diff
@@ -0,0 +1,89 @@
+package de.tum.in.www1.artemis.lecture.service.messaging;
+
+import de.tum.in.www1.artemis.config.MessageBrokerConstants;
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.lecture.service.LectureService;
+import de.tum.in.www1.artemis.lecture.service.LectureUnitService;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.dto.UserLectureDTO;
+import de.tum.in.www1.artemis.service.util.JmsMessageUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jms.annotation.EnableJms;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.stereotype.Component;
+
+import javax.jms.Message;
+import java.util.Set;
+
+/**
+ * Message broker consumer to consume messages coming from Artemis
+ */
+@Component
+@EnableJms
+public class ArtemisConsumer {
+    private final Logger log = LoggerFactory.getLogger(ArtemisConsumer.class);
+
+    private final JmsTemplate jmsTemplate;
+
+    private final LectureUnitService lectureUnitService;
+
+    private final LectureService lectureService;
+
+    public ArtemisConsumer(JmsTemplate jmsTemplate, LectureUnitService lectureUnitService, LectureService lectureService) {
+        this.jmsTemplate = jmsTemplate;
+        this.lectureUnitService = lectureUnitService;
+        this.lectureService = lectureService;
+    }
+
+    /**
+     * Consume message, filter active lecture attachments and send response
+     *
+     * @param message the message to consume
+     */
+    @JmsListener(destination = MessageBrokerConstants.LECTURE_QUEUE_FILTER_ACTIVE_ATTACHMENTS)
+    public void filterLectureActiveAttachments(Message message) {
+        log.debug("Received message in queue {} with body {}", MessageBrokerConstants.LECTURE_QUEUE_FILTER_ACTIVE_ATTACHMENTS, message.toString());
+        UserLectureDTO userLectureDTO = JmsMessageUtil.parseBody(message, UserLectureDTO.class);
+        SecurityUtils.setAuthorizationObject();
```

Comment:
I don't really know how this JmsListener works. Do you have documented that somewhere how we use that?
Because what directly catches my eye, is that we have some input and without any checks set an authorization object.

Why is that okay from a security point of view?

Reply:
You need to authenticate yourself in order to add messages to the message broker. The message broker has a username and password which is configured in each application and used by it when connecting to the broker. Therefore, I cannot connect to the broker without being authenticated. We do the same in InstanceMessageReceiveService when we communicate between nodes. Similarly to that, user data is not added in this communication, which requires the need to set the authorization in this way. Since you need to authenticate in order to connect to the message broker, I think it's okay from the security point of view. What do you think?

Reply:
I'm not 100% sure if I understand that but basically, you say that these methods are not exposed to the outside but to the internal network of nodes and MS?

Reply:
Yes, exactly. Those methods listen for messages in a specific queue in the message broker. The message broker itself is not exposed to the outside world + any node from the internal network accessing it should authenticate in order to be able to read or write messages